Config (still) has private fields in `0.26`

See #841 .

This is still broken in `0.26`. This makes `Builder::with_config()` a PITA to use.

```rust
    cbindgen::Builder::new()
        .with_crate(std::env::var("CARGO_MANIFEST_DIR").unwrap())
        .with_config(cbindgen::Config {
            language: cbindgen::Language::C,
            braces: cbindgen::Braces::SameLine,
            cpp_compat: true,
            style: cbindgen::Style::Both,
            ..Default::default()
        })
        .generate()?
        .write_to_file("binding.h");
```

```
error[E0451]: field `config_path` of struct `Config` is private
   --> foo/build.rs:104:15
    |
104 |             ..Default::default()
    |               ^^^^^^^^^^^^^^^^^^ field `config_path` is private
```

--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,7 @@
+# unreleased
+
+      * Revert: The `Config` struct now has a private member.
+
 # 0.26.0
 
       * Fix swapping of `>>=` and `<<=` in constants.

--- a/src/bindgen/config.rs
+++ b/src/bindgen/config.rs
@@ -1010,8 +1010,15 @@ pub struct Config {
     pub only_target_dependencies: bool,
     /// Configuration options specific to Cython.
     pub cython: CythonConfig,
+    #[doc(hidden)]
     #[serde(skip)]
-    pub(crate) config_path: Option<StdPathBuf>,
+    /// Internal field for tracking from which file the config was loaded.
+    ///
+    /// Users should not set this field explicitly. Making the field private
+    /// prevents users from filling the struct with `..Default::default()`,
+    /// and creating a new InternalConfig struct would require more breaking
+    /// changes to our public API.
+    pub config_path: Option<StdPathBuf>,
 }
 
 impl Default for Config {

